pkg/service/backup/restore_worker_data.go

```diff
@@ -0,0 +1,732 @@
+			if len(takenIDs) > 0 {
+				return takenIDs
+			}
+			// Here host hasn't taken any IDs and pool is empty,
+			// so it waits for the whole table to be restored or
+			// for IDs that might return to the pool in case of error on the other hosts.
+			select {
+			case id, ok := <-w.bundleIDPool:
+				if !ok {
+					return takenIDs
+				}
+				takenIDs = append(takenIDs, id)
+			case <-ctx.Done():
+				return takenIDs
+			}
+		}
+	}
+
+	return takenIDs
+}
+
+// waitJob waits for rclone job to finish while updating its progress.
+func (w *restoreWorker) waitJob(ctx context.Context, pr *RestoreRunProgress) (err error) {
+	defer func() {
+		// On error stop job
+		if err != nil {
+			w.Logger.Info(ctx, "Stop job", "host", pr.Host, "run_progress", *pr, "error", err)
+			if e := w.Client.RcloneJobStop(context.Background(), pr.Host, pr.AgentJobID); e != nil {
+				w.Logger.Error(ctx, "Failed to stop job",
+					"host", pr.Host,
+					"id", pr.AgentJobID,
+					"error", e,
+				)
+			}
+		}
+		// On exit clear stats
+		if e := w.clearJobStats(context.Background(), pr.AgentJobID, pr.Host); e != nil {
+			w.Logger.Error(ctx, "Failed to clear job stats",
+				"host", pr.Host,
+				"id", pr.AgentJobID,
+				"error", e,
+			)
+		}
+	}()
+
+	for {
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+
+		job, err := w.Client.RcloneJobProgress(ctx, pr.Host, pr.AgentJobID, w.Config.LongPollingTimeoutSeconds)
+		if err != nil {
+			return errors.Wrap(err, "fetch job info")
+		}
+
+		switch scyllaclient.RcloneJobStatus(job.Status) {
+		case scyllaclient.JobError:
+			return errors.Errorf("job error (%d): %s", pr.AgentJobID, job.Error)
+		case scyllaclient.JobSuccess:
+			w.updateDownloadProgress(ctx, pr, job)
+			w.Logger.Info(ctx, "Batch download completed", "host", pr.Host, "job_id", pr.AgentJobID)
+			return nil
+		case scyllaclient.JobRunning:
+			w.updateDownloadProgress(ctx, pr, job)
+		case scyllaclient.JobNotFound:
+			return errJobNotFound
+		}
+	}
+}
+
+func (w *restoreWorker) updateDownloadProgress(ctx context.Context, pr *RestoreRunProgress, job *scyllaclient.RcloneJobProgress) {
+	pr.DownloadStartedAt = nil
+	// Set StartedAt and CompletedAt based on Job
+	if t := time.Time(job.StartedAt); !t.IsZero() {
+		pr.DownloadStartedAt = &t
+	}
+	pr.DownloadCompletedAt = nil
+	if t := time.Time(job.CompletedAt); !t.IsZero() {
+		pr.DownloadCompletedAt = &t
+	}
+
+	var (
+		deltaDownloaded = job.Uploaded - pr.Downloaded
+		deltaSkipped    = job.Skipped - pr.Skipped
+		deltaFailed     = job.Failed - pr.Failed
+	)
+
+	pr.Error = job.Error
+	pr.Downloaded = job.Uploaded
+	pr.Skipped = job.Skipped
+	pr.Failed = job.Failed
+
+	w.metrics.UpdateFilesProgress(w.ClusterID, pr.ManifestPath, pr.Keyspace, pr.Table,
+		deltaDownloaded, deltaSkipped, deltaFailed)
+
+	w.insertRunProgress(ctx, pr)
+}
+
+func (w *restoreWorker) restoreSSTables(ctx context.Context, host, keyspace, tables string) error {
+	// TODO: how to set the following variables?
```

Comment:
I understand you want to use retry..backoff mechanism, right ?
The question is related to how to achieve the possibility of setting / overriding these default values you set here ?

If so, then there is some work done on `do_not_run_backup_or_repair_when_load_is_high` branch.
I would say it's ok, to commit it as is. (Remove TODO, and apply the fix when the branch I mentioned is merged)

Reply:
Same as comment below.